grep-9999-modtool-template-rework.md

```diff
@@ -0,0 +1,68 @@
+# GREP [XXXX] -- [Rework gr_modtool's templating]
+
+- Original Author: Marcus Müller <mmueller@gnuradio.org>
+- Champion: Marcus Müller <mmueller@gnuradio>
+- Status: Draft
+
+History:
+- 2021-11-29: Initial Draft
+
+## Abstract
+
+Untangle modtool's templates from strings embedded in Python code, rendering of
+old-style format strings (`"Copyright %s" % datetime.now().year`), Mako
+templates and manual replacements. Get rid of cruft.
+
+## Copyright / License
+
+ GREP "Rework gr_modtool's templating" by Marcus Müller is licensed under CC BY
+ 4.0. To view a copy of this license, visit
+ http://creativecommons.org/licenses/by/4.0/
+ 
+## Motivation
+
+GNU Radio's modtool has quite some history and has come from the need to
+automate the copying of a 3.5/3.6-era "square_ff" howto OOT to a comprehensive
+tool that does much more, including functionality to parse and generate program
+code.
+
+It also had to go through a Python version change, be migrated away from
+Cheetah, and grew alongside with its developers.
+
+As a result, many provisionary solutions became state of the art. Over time,
+this introduced a couple subtle and not-so-subtle bugs.
+
+A cleaner structure for templates should not only simplify development, but also
+remove the likelihood of mistakes.
+
+## Description
+
+### Cruft Removal
+
+Remove all pre-3.8 templating that would need to be reworked.
```

Comment:
I would even go so far as to say pre-3.9.  3.9 was a hard break in modtool, so there is no way with a 3.9 installation to manage an OOT from 3.8 or earlier.  But 3.10 should be able to manage both 3.9 and 3.10